torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -692,11 +688,16 @@ at::Tensor XLANativeFunctions::atan2(const at::Tensor& self,
     return at::native::call_fallback_fn<&xla_cpu_fallback,
                                         ATEN_OP(atan2)>::call(self, other);
   }
-  return DoBinaryOp(self, other,
-                    [&](const XLATensorPtr& xself, const XLATensorPtr& xother,
-                        at::ScalarType dtype) {
-                      return XLATensor::atan2(xself, xother, dtype);
-                    });
+
+  auto common_device = torch_xla::bridge::GetXlaDevice(self, other);
+  XLA_CHECK(common_device);
+  torch::lazy::NodePtr node = torch::lazy::MakeNode<Atan2>(
+      bridge::GetXlaTensor(self)->GetIrValue(),
+      bridge::GetXlaTensor(other)->GetIrValue(),
+      bridge::GetXlaTensor(at::result_type(self, other))->GetIrValue());
```

Comment:
The `MakeNode` calls the constructor for `Atan2` node defined at `LazyIr.h`:
```
Atan2(const torch::lazy::Value& self, const torch::lazy::Value& other)
```
It should only take `self` and `other` tensors as parameters.